### PR TITLE
Update constants v0.6

### PR DESCRIPTION
**Pull request by 0xKiwi**
Part of #2307 

Changes include:
WhistlerBlowerRewardQuotient -> WhistleBlowingRewardQuotient

Added: 
MaxIndicesPerAttestation
MinPerEpochChurnLimit
ShuffleRoundCount
HighBalanceIncrement
SlotsPerEth1VotingPeriod
SlotsPerHistoricalRoot  
MinValidatorWithdrawalDelay 
PersistentCommitteePeriod 
MaxCrosslinkEpochs     
MaxTransfers
DomainTransfer
ProposerRewardQuotient 
MinPenaltyQuotient

Deprecated:
BeaconChainShardNumber
MaxIndicesPerSlashableVote
MaxExitDequeuesPerEpoch
ForkChoiceBalanceIncrement
GenesisForkVersion
GenesisStartShard
EmptySignature (Removed from the spec, should we keep this since we don't have sigs?)
EpochsPerEth1VotingPeriod
LatestBlockRootsLength
AttestationInclusionRewardQuotient

### CODE REVIEW

File: shared/params/config.go [hunk 1]
```diff
@@ -9,81 +9,112 @@ import (
...
-	MaxDepositAmount           uint64 // MaxDepositAmount is the maximal amount of Gwei a validator can send to the deposit contract at once.
-	EjectionBalance            uint64 // EjectionBalance is the minimal GWei a validator needs to have before ejected.
+	// Gwei value constants.
+	MinDepositAmount     uint64 // MinDepositAmount is the maximal amount of Gwei a validator can send to the deposit contract at once.
+	MaxDepositAmount     uint64 // MaxDepositAmount is the maximal amount of Gwei a validator can send to the deposit contract at once.
+	EjectionBalance      uint64 // EjectionBalance is the minimal GWei a validator needs to have before ejected.
+	HighBalanceIncrement uint64 // HighBalanceIncrement is used for converting the high balance into the low balance for validators.
+	// TODO(2307): Remove deprecated fields
 	ForkChoiceBalanceIncrement uint64 // ForkChoiceBalanceIncrement is used to track block score based on balances for fork choice.
+	//Deprecated: Do not use.
```
Comment (prestonvanloon) [inline shared/params/config.go]: What does this go to?

Comment (0xKiwi) [reply shared/params/config.go]: This was changed in 
https://github.com/ethereum/eth2.0-specs/commit/0759e170a7faafca0cee7b9b8929b0c6e15bcf77

Comment (prestonvanloon) [reply shared/params/config.go]: Then lets remove it? it is a new line of code that does to nothing

File: shared/params/config.go [hunk 2]
```diff
@@ -9,81 +9,112 @@ import (
...
+	ShuffleRoundCount        uint64 // ShuffleRoundCount is used for retrieving the permuted index.
+	// TODO(2307): Remove deprecated fields
+	MaxBalanceChurnQuotient uint64 // MaxBalanceChurnQuotient is used to determine how many validators can rotate per epoch.
+	// Deprecated: No longer in use.
+	BeaconChainShardNumber uint64 // BeaconChainShardNumber is the shard number of the beacon chain.
+	// Deprecated: Do not use.
+	MaxIndicesPerSlashableVote uint64 // MaxIndicesPerSlashableVote is used to determine how many validators can be slashed per vote.
+	// Deprecated: Do not use.
+	MaxExitDequeuesPerEpoch uint64 // MaxWithdrawalsPerEpoch is the max withdrawals can happen for a single epoch.
+	// Deprecated: Do not use.
```
Comment (prestonvanloon) [inline shared/params/config.go]: this one too

**Pull request merged**